### ISSUE TITLE
"error  'Set' is not defined  no-undef" in ecmaVersion 2017

### ISSUE THREAD

**Issue by JTronLabs** [eslint/eslint#7984]
**Tell us about your environment**

* **ESLint Version:**
└─┬ gulp-eslint@3.0.1
  └── eslint@3.12.2

* **Node Version:**
v4.2.6

* **npm Version:**
npm: '3.5.2'

**What parser (default, Babel-ESLint, etc.) are you using?**

gulp-eslint

**Please show your full configuration:**

```

env:
  browser: true                       # Enable Browser global variables

parserOptions:
  ecmaVersion: 2017                   # Validate ECMAScript 2016 features
  sourceType: module                  # Allow ECMAScript modules

globals:
  PIXI: false
  Phaser: false

extends: 'eslint:recommended'

rules:
  indent: [2, 2]                      # Indent code with 2 spaces
  quotes: [2, single]                 # Use single quotes for strings
  linebreak-style: [2, unix]          # Favor Unix-style line endings
  semi: [2, always]                   # End lines with semicolons
```

**What did you do? Please include the actual source code causing the issue.**

```js
  static getLockedBirds(game){
    var allBirdIds=new Set();
    for(let i=0;i<=game.animationInfo.maxBirdFrame;i++){
      allBirdIds.add(i);
    }

    var unlockedBirds = new Set(DataAccess.getConfig('unlockedBirdSprites'));
    var lockedBirds = [...allBirdIds].filter(x => !unlockedBirds.has(x)); //find all bird ids that are not in the unlocked set but ARE in the allBird set

    return lockedBirds;
  }
```

**What did you expect to happen?**
Set is a ECMA 2015 object. By using ECMA 8/2017 I expected the parser to recognize this. If I add `es6: true` to the environment configuration, no error is thrown

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/home/james/Documents/birdu/src/objects/Helpers/DataAccess.js
  31:24  error  'Set' is not defined  no-undef
  36:29  error  'Set' is not defined  no-undef
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

The `parserOptions` config only handles options for the parser (e.g. if you're using ES6 *syntax*). However, it doesn't affect the list of availale globals, which is used by rules like `no-undef`. To set ES6 globals, you should add `env: {es6: true}` to your config (as you mentioned you did).

**Comment by platinumazure** [comment]
We get this question with some frequency. I think this would make a good FAQ item in our README. @not-an-aardvark What do you think?

**Comment by not-an-aardvark** [comment]
I'm not against the idea, but I'm also not convinced that it's necessary given that we don't even mention `ecmaVersion` in our readme. Maybe we should put a note about `env` when we mention `ecmaVersion` on the [Configuring](http://eslint.org/docs/user-guide/configuring#specifying-parser-options) page.

**Comment by platinumazure** [comment]
Agreed, for some reason I thought we mentioned ecmaVersion but you are right, it links to the Configuring page.

I'd like to see a paragraph on the difference between ecmaVersion (syntax) and envs (globals), yeah.

**Comment by Zanadar** [comment]
@qlaire @ddipanfilo @mmlanders23 and I are doing some open source learning, and we're gonna add this to the FAQ.

**Comment by not-an-aardvark** [comment]
Thanks for contributing!

Instead of the FAQ, I wonder if the [Specifying Parser Options](http://eslint.org/docs/user-guide/configuring#specifying-parser-options) section of the docs would be a better place to note this. The FAQ doesn't go into detail about how to configure parser options at all, I think it would be better to clarify what `parserOptions` does/doesn't do in its dedicated section.
